src/storage/src/hummock/sstable/multi_builder.rs

```diff
@@ -228,6 +229,15 @@ where
                     // default
                     self.largest_vnode_in_current_partition = VirtualNode::MAX.to_index();
                 }
+            } else {
+                self.last_table_id = user_key.table_id.table_id;
+                self.split_weight_by_vnode = 0;
+                self.largest_vnode_in_current_partition = VirtualNode::MAX.to_index();
+                if let Some(builder) = self.current_builder.as_ref()
```

Comment:
If we introduce this logic, it may bring more 4m sst when table_id switches. This seems to be contrary to https://github.com/risingwavelabs/risingwave/pull/16495, which will bring more sst at high level 

PTAL @hzxa21 @zwang28

Reply:
But it is only split by table, not partition vnode. I think it is acceptable.